src/gl/gl3device.cpp

```diff
@@ -24,12 +25,86 @@ namespace gl3 {
 GlGlobals glGlobals;
 
 Gl3Caps gl3Caps;
-// terrible hack for GLES
 bool32 needToReadBackTextures;
 
 int32   alphaFunc;
 float32 alphaRef;
 
+#if defined(LIBRW_SDL2) || defined(LIBRW_SDL3)
+// Helper function to log OpenGL info using SDL logging and optionally to file
+static void
+logGLInfo(void)
+{
+	const char *version     = (const char *)glGetString(GL_VERSION);
+	const char *vendor      = (const char *)glGetString(GL_VENDOR);
+	const char *renderer    = (const char *)glGetString(GL_RENDERER);
+	const char *extensions  = (const char *)glGetString(GL_EXTENSIONS);
+	const char *shadingLang = (const char *)glGetString(GL_SHADING_LANGUAGE_VERSION);
+
+	GLint maxVertexAttribs, maxVertexUniforms, maxFragmentUniforms, maxTextureSize;
+	glGetIntegerv(GL_MAX_VERTEX_ATTRIBS, &maxVertexAttribs);
+	glGetIntegerv(GL_MAX_VERTEX_UNIFORM_VECTORS, &maxVertexUniforms);
+	glGetIntegerv(GL_MAX_FRAGMENT_UNIFORM_VECTORS, &maxFragmentUniforms);
+	glGetIntegerv(GL_MAX_TEXTURE_SIZE, &maxTextureSize);
+
+	// Always log to SDL_Log (shows in logcat on Android)
+	SDL_Log("== OpenGL Info ==");
+	SDL_Log("GL_VERSION: %s", version ? version : "N/A");
+	SDL_Log("GL_VENDOR: %s", vendor ? vendor : "N/A");
+	SDL_Log("GL_RENDERER: %s", renderer ? renderer : "N/A");
+	SDL_Log("GL_SHADING_LANGUAGE_VERSION: %s", shadingLang ? shadingLang : "N/A");
+	SDL_Log("GL_MAX_VERTEX_ATTRIBS: %d", maxVertexAttribs);
+	SDL_Log("GL_MAX_VERTEX_UNIFORM_VECTORS: %d", maxVertexUniforms);
+	SDL_Log("GL_MAX_FRAGMENT_UNIFORM_VECTORS: %d", maxFragmentUniforms);
+	SDL_Log("GL_MAX_TEXTURE_SIZE: %d", maxTextureSize);
+
+	// Try to also write to file in application directory
+#if defined(LIBRW_SDL2)
```

Comment:
Personally, I'd find an application/game that unconditionally creates a log file annoying. Logging to stdout should be enough. Users can save it themselves by, or through a configurable option.